portability-transfer/src/main/java/org/datatransferproject/transfer/JobPollingService.java

```diff
@@ -106,8 +112,9 @@ private boolean tryToClaimJob(UUID jobId, KeyPair keyPair) {
       logger.debug("public key cannot be persisted again");
       return false;
     }
-    // Populate job with public key to persist
-    String encodedPublicKey = BaseEncoding.base64Url().encode(keyPair.getPublic().getEncoded());
+
+    String kid = UUID.randomUUID().toString();
+    JWK jwk = new RSAKey.Builder((RSAPublicKey) keyPair.getPublic()).keyID(kid).build();
```

Comment:
Per discussion the other day, were you able to look into extracting the jose logic to a new class, if it's simple enough, so we don't have to pull in the dependnecy in our backend?

If not, could you leave a TODO?